spec/integration/unknown_options_spec.rb

```diff
@@ -0,0 +1,39 @@
+require "open3"
+require "pact_broker/client/cli/broker"
+
+# This is not the ideal way to write a test, but I tried to write it with an in memory invocation,
+# and I couldn't get the capture to work, and it became super complicated.
+
+RSpec.describe "using unknown options", skip_windows: true do
+  let(:unknown_switches_text) { "Unknown switches" }
+  let(:warning_text) { "This is a warning"}
+  let(:command) { "bundle exec bin/pact-broker can-i-deploy --pacticipant Foo --foo --broker-base-url http://example.org" }
+
+  it "prints an 'unknown switches' warning to stderr and also includes the normal output of the command" do
+    stderr_lines = nil
+
+    Open3.popen3(command) do |stdin, stdout, stderr, thread|
+       stderr_lines = stderr.readlines
+    end
+
+    expect(stderr_lines.join("\n")).to include(unknown_switches_text)
```

Comment:
it's a shame it says unknown switches rather than unknown Option but this is still a much better experience

>Usage:
  pact-broker publish PACT_DIRS_OR_FILES ... -a, --consumer-app-version=CONSUMER_APP_VERSION -b, --broker-base-url=BROKER_BASE_URL

>Options:
  -a, --consumer-app-version=CONSUMER_APP_VERSIO

Reply:
Yes, the terminology used inside the gem is very inconsistent. The words argument, option and switch are all used to mean the same thing in various places. I guess it's what happens when you have multiple people working on a project and you don't have a Terminology Dictator in control.